pandas/_libs/hashtable_func_helper.pxi.in

```diff
@@ -77,54 +77,74 @@ cdef build_count_table_{{dtype}}(const {{dtype}}_t[:] values,
 @cython.wraparound(False)
 @cython.boundscheck(False)
 {{if dtype == 'object'}}
-cpdef value_count_{{dtype}}(ndarray[{{dtype}}] values, bint dropna):
+cpdef value_count_{{dtype}}(ndarray[{{dtype}}] values, bint dropna, navalue=np.NaN):
 {{else}}
 cpdef value_count_{{dtype}}(const {{dtype}}_t[:] values, bint dropna):
 {{endif}}
     cdef:
         Py_ssize_t i = 0
+        Py_ssize_t n = len(values)
+        size_t unique_key_index = 0
+        size_t unique_key_count = 0
         kh_{{ttype}}_t *table
 
-        {{if dtype != 'object'}}
-        {{dtype}}_t[:] result_keys
-        int64_t[:] result_counts
-        {{endif}}
-
         # Don't use Py_ssize_t, since table.n_buckets is unsigned
         khiter_t k
+        bint is_null
 
-    table = kh_init_{{ttype}}()
-    {{if dtype == 'object'}}
-    build_count_table_{{dtype}}(values, table, 1)
+        {{c_type}} val
+
+        int ret = 0
+
+    {{if dtype[0]!='u'}}
+    result_keys = {{dtype.title()}}Vector()
     {{else}}
-    build_count_table_{{dtype}}(values, table, dropna)
+    result_keys = {{'U'+dtype[1::].title()}}Vector()
     {{endif}}
-
-    result_keys = np.empty(table.n_occupied, '{{dtype}}')
-    result_counts = np.zeros(table.n_occupied, dtype=np.int64)
+    result_counts = Int64Vector()
+    table = kh_init_{{ttype}}()
 
     {{if dtype == 'object'}}
-    for k in range(table.n_buckets):
-        if kh_exist_{{ttype}}(table, k):
-            result_keys[i] = <{{dtype}}>table.keys[k]
-            result_counts[i] = table.vals[k]
-            i += 1
+    kh_resize_{{ttype}}(table, n // 10)
```

Comment:
why this?

Reply:
I kept it as it was in the original version. One probably should extend `{{name}}HashTable` interface and use it instead of c-function directly - it has a resize built already in: https://github.com/pandas-dev/pandas/blob/f1d4026973bfb650be63cbe87443570aa01935dd/pandas/_libs/hashtable_class_helper.pxi.in#L400-L404